libraries/botbuilder-adapter-twilio-whatsapp/src/twilioWhatsAppAdapter.ts

```diff
@@ -57,24 +94,51 @@ export class TwilioWhatsAppAdapter extends BotAdapter {
     protected readonly client: Twilio.Twilio;
     protected readonly channel: string = 'whatsapp';
 
+    // OAuth Properties
+    protected readonly credentials: AppCredentials;
+    protected readonly credentialsProvider: SimpleCredentialProvider;
     /**
      * Creates a new TwilioWhatsAppAdapter instance.
      * @param settings configuration settings for the adapter.
      */
-    public constructor(settings: TwilioWhatsAppAdapterSettings) {
+    public constructor(settings?: Partial<TwilioWhatsAppAdapterSettings>) {
```

Comment:
What about adding a second parameter to the constructor, which is the default `BotFrameworkAdapterSettings`? Or is this too much overhead?

Would be great if we could use the default adapter interface, as an optional constructor parameter.

Reply:
I tried replicating what the BotFrameworkAdapter does with the settings parameter.

As for adding an optional flag or property, I don't really see the need for it, normally if the user does not provide any OAuth settings and does not call any OAuthPrompt that part of the code should never be called.

> Would be great if we could use the default adapter interface, as an optional constructor parameter.

I don't understand what you mean by that, I do realize however that I forgot to implement the `IUserTokenProvider` interface in the pull request.

Reply:
```suggestion
    public constructor(settings: TwilioWhatsAppAdapterSettings, botFrameworkAdapterSettings?: BotFrameworkAdapterSettings ) {
```

I meant using the official BotFrameworkAdapterSettings as parameter for the constructor, so it is really easy to combine multiple adapters in a project. And the structure of `BotFrameworkAdapterSettings` has the required keys for the OAuth settings.

Reply:
What do you think of doing something like this instead?
```ts
export interface TwilioWhatsAppAdapterSettings extends BotFrameworkAdapterSettings {

    accountSid: string;

    authToken: string;

    phoneNumber: string;

    endpointUrl: string;
}

Reply:
For me, this is too highly coupled, what if the `BotFrameworkAdapterSettings` implements a key that is also present in our `TwilioWhatsAppAdapterSettings` for example. (and some more concerns).

My idea was to port your code to CustomWebAdapter, so your work can be leveraged in every adapter (and every adapter to come). 

If we keep the Twilio and BF adapter settings separate, we will benefit from:
1. Developers using multiple adapters (like the Bot Framework adapter and Twilio) can reuse the same settings object.
2. Custom Adapter developers can extend the `CustomWebAdapter` and pass the optional `BotFrameworkAdapterSettings` via `super()` if they would like to allow OAuth.

I am attending a conference this week, so my spare time is really limited, but I could help you port it to the CustomWebAdapter / make a draft of my proposal above.

What do you think @nestoralonsovina / @szul?

Reply:
Here a quick example of how it could be combined in the `CustomWebAdapter`; [README](https://github.com/iMicknl/botbuilder-community-js/tree/core-work%2Boauth/libraries/botbuilder-core#customwebadapter) + [CODE](https://github.com/iMicknl/botbuilder-community-js/blob/core-work%2Boauth/libraries/botbuilder-core/src/customWebAdapter.ts)

Reply:
I understand your point now, everything seems ok for me.